File: BDH.Rhino.Web.API.Domain/Entities/BuildingConcept.cs
using BDH.Rhino.Web.API.Domain.Bouwkosten;
using System.ComponentModel.DataAnnotations;

namespace BDH.Rhino.Web.API.Domain.Entities
{
    public class BuildingConcept
    {
        [Key]
        public Guid Id { get; set; }
        public string Name { get; set; } = null!;


        public decimal BvoPerUnit { get; set; }
        public decimal M3PerUnit { get; set; }
        public int WoningenPerUnit { get; set; }
        public decimal ProductieKostenPerUnit { get; set; }
        public decimal MeerprijsEPC { get; set; }
        public decimal MeerprijsBENG { get; set; }
        public bool Stapelbaar { get; set; } = false;
        public bool IsPrivate { get; set; } = true;
        public double Width { get; set; }
        public double Depth { get; set; }
        public double Height { get; set; }
        public double GlazingFactor { get; set; }


        public BuildingConceptGeometry Geometry { get; set; } = null!;
        public User CreatedBy { get; set; } = null!;
        public Company Owner { get; set; } = null!;
        public BuildingConceptCatalog Catalog { get; set; } = null!;

        public decimal BouwkostenPerBVO => BvoPerUnit == 0 ? 0 : ProductieKostenPerUnit / BvoPerUnit;


        public TypologieKostenModel ToTypology()
        {
            return new TypologieKostenModel(Name, Id.ToString(), BouwkostenPerBVO, BvoPerUnit, false, WoningenPerUnit, MeerprijsEPC, MeerprijsBENG);
        }
    }
}

File: BDH.Rhino.Web.FrontEnd/src/app/data/schema/models/BuildingConcept.ts
import { DataModel } from "@app/modules/project-editor/models/DataModel";

export class BuildingConcept extends DataModel {
    constructor( 
        public id: string,
        public name: string,
        public createdBy: string,
        public company: string,
        public bvoPerUnit: number,
        public m3PerUnit: number,
        public width: number,
        public depth: number,
        public height: number,
        public glazingFactor : number,
        public woningenPerUnit: number,
        public productieKostenPerUnit: number,
        public isPrivate: boolean,
        public isFavorited: boolean){
            super();
        }
}
